analyzers/copyboundary/copy_boundary.go

```diff
@@ -0,0 +1,100 @@
+// Package copyboundary defines an Analyzer that checks for
+// uses of copy of slice & map.
+// https://github.com/uber-go/guide/blob/master/style.md#copy-slices-and-maps-at-boundaries
+package copyboundary
+
+import (
+	"go/ast"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+const Doc = `check for uses of copy of slice & map
+
+This checker reports usages of slice & map that are directly copied. 
+Slices and maps contain pointers to the underlying data so be wary 
+of scenarios when they need to be copied..
+`
+
+var Analyzer = &analysis.Analyzer{
+	Name: "copyboundary",
+	Doc:  Doc,
+	Run:  run,
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	inspector := newInspector(pass)
+	for _, file := range pass.Files {
+		for _, inspect := range inspector.collections {
+			ast.Inspect(file, inspect)
+		}
+	}
+	return nil, nil
+}
+
+type copySliceMapInspector struct {
```

Comment:
@faem  Can we make the name consistent with the analyzer name? Like `copyBoundaryInspector`